### DISCUSSION TITLE
PIMCORE 11: how to setup Gotenberg pdf generator in pimcore web to print

### DISCUSSION THREAD

**Discussion by sandyHM** [pimcore/pimcore#17823] [Q&A / Support] [answered]
how to setup Gotenberg in pimcore web to print
![gotenberg setting pimcore ](https://github.com/user-attachments/assets/8412daf2-ba39-4d6d-ab2e-bdf134401f5a)

**Comment by Dejan-CP** [comment]
I have the same question, especially since the link to the Option Documentation is a dead link

**Reply by kingjia90** [reply]
The new link is https://gotenberg.dev/docs/routes#page-properties-chromium
It will be fixed in the bundle by https://github.com/pimcore/web-to-print-bundle/pull/90 and released next Tuesday

Thank you again for letting us know about the dead link

**Reply by Dejan-CP** [reply]
Thanks for the reply! 
About the header and footer: The documentation states how to do it with cURL, e.g., 
`--form files=@/path/to/header.html \
--form files=@/path/to/footer.html \`

How can I pass that information in the Gotenberg Settings Textarea in pimcore? Where is pimcore looking for those html files?

**Reply by kingjia90** [reply]
The settings textarea needs to be a valid json array.

AFAIK not everything in the documentation is fully supported, we are more subject to https://github.com/gotenberg/gotenberg-php

You can see more on how it get processed and which value are actually supported in https://github.com/pimcore/web-to-print-bundle/blob/d18a09bc07179b3bc14dc6e923dd1fdb793c6841/src/Processor/Gotenberg.php#L88

For the header and footer, by looking at https://github.com/pimcore/web-to-print-bundle/blob/d18a09bc07179b3bc14dc6e923dd1fdb793c6841/src/Processor/Gotenberg.php#L157 which just ends up to a `fopen` a stream of the filename given, you can put the files anywhere in the local directory as long it can be reached by an absolute path

Hope it helps

**Comment by sandyHM** [comment]
@kingjia90 Thank you for prompt reply .
I have another question, Is there any installer of Gotenberg or can we use Gotenberg without docker and docker-compose ?

**Reply by GNi33** [reply, marked as answer]
Gotenberg itself is shipped (afaik) as a docker image only (https://gotenberg.dev/docs/getting-started/introduction)
